### PR TITLE
Workspace Feed - add Search for more than 8 export rows

### PR DESCRIPTION
**Pull request by narefyev91**
### Explanation of Change
Add missing search input - when user has more than 8 export items to choose in Company card detail view

### Fixed Issues

$ https://github.com/Expensify/App/issues/51884
PROPOSAL:

### Tests
1. Go to staging.new.expensify.com
2. Choose the workspace from settings
3. Click company cards >Card Details
4. Click Quickbooks Online Debit card Export
5. Verify that it will be no search - if less than 8 export items presented
6. Verify that it will be a search - if 8+ export items presented
7. Verify that search functionality is working

- [x] Verify that no errors appear in the JS console

### Offline tests
No changes here

### QA Steps
1. Go to staging.new.expensify.com
2. Choose the workspace from settings
3. Click company cards >Card Details
4. Click Quickbooks Online Debit card Export
5. Verify that it will be no search - if less than 8 export items presented
6. Verify that it will be a search - if 8+ export items presented
7. Verify that search functionality is working

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/fd8c4d68-de9c-4e09-866e-8662d3eef44a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/57d0378b-2c34-4f0b-812b-a6efeee560f4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/0be09994-ed4c-4eec-b35c-322c596058e1

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/6118f844-f80d-4b0a-8b0a-a6b398e375fe

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/0430da5d-8e91-4213-9f81-6aea74bc2431

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/78d0343f-0b9e-47d6-a45b-05670c5cb38a

</details>

### CODE REVIEW

**Comment by narefyev91** [comment]
> @narefyev91 can you please provide me mock data for cardlist:)

use this useEffect:

>     useEffect(() => {
        Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {
            connections: {
                quickbooksOnline: {
                    config: {
                        companyName: 'test',
                        credentials: {
                            access_token:
                                '2LHOKnH9waofJaMRyrVBUP+4qJsh4u+6E3Ffu73YJFJ6FnzAfsydZVkVFwjQmyxknEXCOYNeqoRljafcUk2sgzeq6002fXY55AEKF36UjaQQskfo4wR192K0v/68BUEzCpghfahoRmupoNq0u1JM8h/NDNIcLEi9ZDBno+7a22wi2jshJtcGL/wGBMUalWFupOmkiJbLPFen+mHFqLI8kF4TydI2a7SC4M4BAkZ2N7cFvp9LC39cLvrL7pZ/0wdIe8d4AMrKdR8M8V+KD8r2X4zXoQhKxSPA+2wRFiXIwOKB2iZqgI19dQHD0QqwbW6a512nOb7Q1Y0/SrF6NtVAaBXwhF2o3VULwFH3cy4QTYxB25wXCU76bhYsvUaQ/xWUVjY2TEi3GXkdTrNN8Ys8MXH3A9kVX7I32whBMfygcjYLtxc4/aVr8gYg8w5XPRRJ5Q/OGy68mX1JlCyWi+1NNAUJVkeRXuOsnivYvvaRisCJ4bng4ZiK9hnSWJ6qyKMWnb/nPpjZfZe25nbXfEG/e3CgwJ3KU4RlWZM+ECSWGcM/KyAGfErbcDBXXQk/6BnMjTqQkP5HotH8BHsUzYsw8D1A8i3gpgU/CStK8b+2PFcoqgTxWmsPgL7O32AFNR6z1yd1Zsrk5cDVHDlTddw1ePxrvfJRuT7gNKlut9oIAjgyR5iagxk/By2AGp0o2S53becAoPRGtuYYxk5WmpfB2q0N0QHI9VtRDJ5p1wn4KPO4twnnkU5YyD/u2UuBCWw9UB1rDWZVpGRUilyF2/0vy7PJHoUCXodz0DIulT+cDSHqf9hmc1pm7LI3Qt17X4uO1YCIQA7qQk1G5aY32HwNVq2VnyZJT64nEJ3JMydI4k74jrGy9Zp+52lAF39AenCaNjbVJg5svd0KEeOdwqr1j9SNkrHZkMmhuAUSI/WfyAhrtmjmRVGBTJ6b9l8fjdzwLwubXdFInG8M7h8Pc0Db7VTvd8Qv8MFRjhomzHbTkVMApKr1EH9WGzf932FlDse9HwKC4c6mA2Uj0BCaxsbnKnqd/I6Ug==;vJoXwrJgPzNOdv28yQHtPA==;eKPhE8t3aDV62W1np/MZoKIYA1bRawdFuttb5yxT44HQWjYgEc5x/Glx+Ye6YZP06Aj3N0A7eMw6rj98Kjpaow==',
                            companyID: '934145206296',
                            companyName: 'test',
                            expires: 1730725016,
                            realmId: '934145340296',
                            refresh_token:
                                'KwizVqdImwI5kHeImQ2NrXO7rteCSzvtQakR3uhp7Y04X06z3F4Hvt42jw==;fK/XjysVFPC4lYgRXbzi/RpxI1z3aWWpaLSAnT5v1kRtg4ilXX9fjQvA6Wc3LhKqZXnADugm2DbveEmZWAItlg==',
                            scope: 'Accounting',
                            token_type: 'bearer',
                        },
                        realmId: '9341453405206',
                        autoSync: {
                            enabled: true,
                            jobID: '4604867689969023701',
                        },
                        syncClasses: 'REPORT_FIELD',
                        pendingFields: {},
                        errorFields: {},
                        syncLocations: 'NONE',
                        autoCreateVendor: false,
                        collectionAccountID: '29',
                        enableNewCategories: true,
                        export: {
                            exporter: 'zhenja3033@gmail.com',
                        },
                        exportDate: 'REPORT_EXPORTED',
                        hasChosenAutoSyncOption: true,
                        lastConfigurationTime: 1730721696740,
                        markChecksToBePrinted: false,
                        nonReimbursableBillDefaultVendor: 'NONE',
                        nonReimbursableExpensesAccount: {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents',
                        },
                        nonReimbursableExpensesExportDestination: 'debit_card',
                        reimbursableExpensesAccount: {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents',
                        },
                        reimbursableExpensesExportDestination: 'bill',
                        reimbursementAccountID: '29',
                        syncCustomers: 'TAG',
                        syncItems: false,
                        syncPeople: false,
                        syncTax: false,
                    },
                    data: {
                        accountPayable: [],
                        accountsReceivable: [],
                        bankAccounts: [
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '29',
                                name: 'Cash and cash equivalents',
                            },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents1',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents2',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents3',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents4',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents5',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents6',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents7',
                          },
                          {
                            currency: 'PLN',
                            glCode: '',
                            id: '29',
                            name: 'Cash and cash equivalents8',
                          },
                        ],
                        country: 'PL',
                        creditCards: [],
                        edition: 'QuickBooks Online Simple Start',
                        employees: [],
                        homeCurrency: 'PLN',
                        isMultiCurrencyEnabled: false,
                        journalEntryAccounts: [
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '32',
                                name: 'Accrued liabilities',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '70',
                                name: 'Allowance for bad debt',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '13',
                                name: 'Available for sale assets (short-term)',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '38',
                                name: 'Dividends payable',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '26',
                                name: 'Income tax payable',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '65',
                                name: 'Inventory',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '74',
                                name: 'Payroll Clearing',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '73',
                                name: 'Payroll liabilities',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '47',
                                name: 'Prepaid expenses',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '71',
                                name: 'Short-term debit',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '1',
                                name: 'Uncategorised Asset',
                            },
                        ],
                        otherCurrentAssetAccounts: [
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '70',
                                name: 'Allowance for bad debt',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '13',
                                name: 'Available for sale assets (short-term)',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '65',
                                name: 'Inventory',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '47',
                                name: 'Prepaid expenses',
                            },
                            {
                                currency: 'PLN',
                                glCode: '',
                                id: '1',
                                name: 'Uncategorised Asset',
                            },
                        ],
                        vendors: [],
                    },
                    lastSync: {
                        errorDate: '',
                        errorMessage: '',
                        isAuthenticationError: false,
                        isConnected: true,
                        isSuccessful: true,
                        source: 'NEWEXPENSIFY',
                        successfulDate: '2024-11-04T12:01:36+0000',
                    },
                },
            },
        });
    }, [policyID]);

**Review by allgandalf — APPROVED** [review]
LGTM and tests well

File: src/CONST.ts [hunk 1]
```diff
@@ -3056,6 +3056,7 @@ const CONST = {
     // Character Limits
     FORM_CHARACTER_LIMIT: 50,
     STANDARD_LENGTH_LIMIT: 100,
+    STANDARD_LIST_ITEM_LIMIT: 8,
```
Comment (mountiny) [inline src/CONST.ts:3059]: When adding such const it should be good to also updated all the applicable usages to use it, maybe you can do it in another PR

**Review by mountiny — APPROVED** [review]
Thanks!

File: src/CONST.ts (hunk 1 above)
Comment (narefyev91) [reply src/CONST.ts:3059]: Yeah - today will add it with upcoming PR.

**Comment by joekaufmanexpensify** [comment]
### Search bar with 8+ rows 
Confirmed search functionality working too. 
image

### No search bar <8 rows 

image
